synapse/lib/storm.py

```diff
@@ -1659,6 +1660,46 @@ async def __anit__(self, query, snap, opts=None, user=None, root=None):
         self._loadRuntVars(query)
         self.onfini(self._onRuntFini)
 
+    async def emitter(self):
+
+        self.emitq = asyncio.Queue(maxsize=1)
+        async def fill():
+            try:
+                async for item in self.execute():
+                    await asyncio.sleep(0)
+                await self.emitq.put((False, None))
+                await self.fini()
```

Comment:
there is kind of a weird loop here, conceptually.
this function is being run as a schedCoro which eventually calls fini on the base object.
that task (which called fini) would then get cancelled.
the activetasks are cancelled before fini functions are called & fevt is set.

need to think on that.

Reply:
simplified